src/client/components/dropdown_container/view.tsx

```diff
@@ -0,0 +1,97 @@
+import { action } from 'mobx'
+import { observable } from 'mobx'
+import { observer } from 'mobx-react'
+import * as React from 'react'
+import { ComponentType } from 'react'
+
+export type WrappedComponentProps = {
+  onToggle: () => void
+}
+
+export const dropdownContainer = (WrappedComponent: ComponentType<any>) => {
```

Comment:
I am not that familiar with HOC in react yet, so this could be standard practice (I think we briefly talked about it). But maybe we could simplify things by just tying `EnhancedDropdown` directly to `Dropdown`? Can you think of a likely scenario where we need to take advantage of the `WrappedComponent` parameter? A bit worried about over-engineering it.